src/UserService.Validation/User/EditUserRequestValidator.cs

```diff
@@ -1,376 +1,283 @@
+          nameof(EditUserRequest.MiddleName),
+          nameof(EditUserRequest.LastName),
+          nameof(EditUserRequest.Status),
+          nameof(EditUserRequest.Rate),
+          nameof(EditUserRequest.City),
+          nameof(EditUserRequest.Gender),
+          nameof(EditUserRequest.DateOfBirth),
+          nameof(EditUserRequest.StartWorkingAt),
+          nameof(EditUserRequest.About),
+          nameof(EditUserRequest.AvatarFileId),
+          nameof(EditUserRequest.IsActive),
+          nameof(EditUserRequest.DepartmentId),
+          nameof(EditUserRequest.RoleId),
+          nameof(EditUserRequest.PositionId),
+          nameof(EditUserRequest.OfficeId),
+        });
+
+      AddСorrectOperations(nameof(EditUserRequest.FirstName), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.MiddleName), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.LastName), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.Status), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.Rate), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.Gender), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.City), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.DateOfBirth), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.StartWorkingAt), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.AvatarFileId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.IsActive), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.DepartmentId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.PositionId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.RoleId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.OfficeId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditUserRequest.About), new List<OperationType> { OperationType.Replace });
 
       #endregion
 
-      #region IsActive
+      #region firstname
+
+      AddFailureForPropertyIf(
+        nameof(EditUserRequest.FirstName),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+        {
+          { x => !string.IsNullOrEmpty(x.value?.ToString().Trim()), "First name must not be empty." },
+          { x => !NameRegex.IsMatch(x.value.ToString()), "First name must not contain numbers." },
```

Comment:
If x.value = null, here will be exception.  may be

Reply:
We use CascadeMode.Stop, if the value is null, the validator will stop at line 75.